libs/core/src/lib/grid-list/components/grid-list-item/grid-list-item.component.ts

```diff
@@ -0,0 +1,343 @@
+        private readonly _elementRef: ElementRef,
+        private readonly _gridListSelectionService: GridListSelectionService<T>
+    ) {
+        const selectedItemsSub = this._gridListSelectionService.selectedItemsObs.subscribe((items) => {
+            this._selectedItem = items.selection.find((item) => item === this.value);
+            this._cd.markForCheck();
+        });
+        this.subscription.add(selectedItemsSub);
+    }
+
+    /** @hidden */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (
+            changes.layoutItemPattern &&
+            changes.layoutItemPattern.previousValue !== changes.layoutItemPattern.currentValue
+        ) {
+            this.gridLayoutClasses = parseItemLayoutPattern(this.layoutItemPattern);
+        }
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        this._cd.detectChanges();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this.subscription.unsubscribe();
+    }
+
+    /** @hidden */
+    singleSelect(event: MouseEvent): void {
+        this._preventDefault(event);
+
+        this._gridListSelectionService.setSelectedItem(this.value, this._index);
+    }
+
+    /** @hidden */
+    _selectionItem(value: number | T): void {
+        const action =
+            this.selectionMode !== 'multiSelect'
+                ? null
+                : value || value === 0
+                ? GridListSelectionActions.ADD
+                : GridListSelectionActions.REMOVE;
+
+        this._gridListSelectionService.setSelectedItem(this.value, this._index, action);
+    }
+
+    /** @hidden */
+    _onDetail(event: MouseEvent): void {
+        this._preventDefault(event);
+
+        this.detail.emit({
+            value: this.value,
+            index: this._index
+        });
+    }
+
+    /** @hidden */
+    _onNavigate(event: MouseEvent): void {
+        this._preventDefault(event);
+
+        this.navigate.emit({
+            value: this.value,
+            index: this._index
+        });
+    }
+
+    /** @hidden */
+    _onDelete(event: MouseEvent): void {
+        this._preventDefault(event);
+
+        this.delete.emit({
+            value: this.value,
+            index: this._index
+        });
+    }
+
+    /** @hidden */
+    _clickOnDraft(event: MouseEvent): void {
+        this._preventDefault(event);
+
+        this.draft.emit({
+            value: this.value,
+            index: this._index
+        });
+    }
+
+    /** @hidden */
+    _clickOnLocked(event: MouseEvent): void {
+        event.stopPropagation();
+        event.preventDefault();
+
+        this.locked.emit({
+            value: this.value,
+            index: this._index
+        });
+    }
+
+    /** @hidden */
+    _onClick(): void {
+        if (this.type !== 'active' && this.type !== 'detailsAndActive') {
+            return;
+        }
+
+        this.press.emit({
+            value: this.value,
+            index: this._index
+        });
```

Comment:
what about using a getter for output event value to simplify and reduce code repeating?
the value seems to be the same for every event
and a getter like this would help:
```
get _outputEventValue(): {
   return {
        value: this.value,
        index: this._index
    }
}
```

```
this.press.emit(this._outputEventValue);
```